reporter/http/http_test.go

```diff
@@ -167,3 +168,91 @@ func TestSpanIsReportedAfterBatchSize(t *testing.T) {
 		t.Errorf("unexpected number of spans received\nhave: %d, want: %d", aNumSpans, eNumSpans)
 	}
 }
+
+func TestSpanCustomHeaders(t *testing.T) {
+	serializer := reporter.JSONSerializer{}
+
+	hc := headerClient{
+		headers: http.Header{
+			"Key1": []string{"val1a", "val1b"},
+			"Key2": []string{"val2"},
+		},
+	}
+	var haveHeaders http.Header
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		haveHeaders = r.Header
+	}))
+	defer ts.Close()
+
+	spans := generateSpans(1)
+
+	rep := zipkinhttp.NewReporter(
+		ts.URL,
+		zipkinhttp.Serializer(serializer),
+		zipkinhttp.Client(hc),
+	)
+	for _, span := range spans {
+		rep.Send(*span)
+	}
+	rep.Close()
+
+	for _, key := range []string{"Key1", "Key2"} {
+		if want, have := hc.headers[key], haveHeaders[key]; !reflect.DeepEqual(want, have) {
+			t.Errorf("header %s: want: %v, have: %v\n", key, want, have)
+		}
+	}
+}
+
+func TestB3SamplingHeader(t *testing.T) {
+	serializer := reporter.JSONSerializer{}
+
+	var haveHeaders map[string][]string
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		haveHeaders = r.Header
+	}))
+	defer ts.Close()
+
+	spans := generateSpans(1)
+
+	rep := zipkinhttp.NewReporter(
+		ts.URL,
+		zipkinhttp.Serializer(serializer),
+		zipkinhttp.AllowSamplingReporterCalls(true),
+	)
+	for _, span := range spans {
+		rep.Send(*span)
+	}
+	rep.Close()
+
+	if len(haveHeaders["B3"]) > 0 {
```

Comment:
I guess golang will send this uppercase?.. where possible we should always do lowercase as it is cheaper in messaging extraction to not have to look twice.

If this is implicilty always the case that first letter is uppercased nevermind me :D

Reply:
Yes it does send it uppercase. I could have casted it back to a http.Header type and then use the Get function but it would have made it even more quircky. Once we drop support for Go <= 1.13 I can refactor to only use http.Header and test will look cleaner:

This is what happens with http header keys:
https://golang.org/src/net/textproto/reader.go#L588

Reply:
thanks.. I always forget this :)